tools/astarte_export/lib/astarte/fetchdata/fetchdata.ex

```diff
@@ -112,49 +154,53 @@ defmodule Astarte.Export.FetchData do
     introspection = device_data.introspection
 
     mapped_interfaces =
-      Enum.reduce(introspection, [], fn {interface_name, major_version}, acc ->
-        {:ok, interface_description} =
-          Queries.fetch_interface_descriptor(conn, realm, interface_name, major_version, [])
-
-        minor_version = interface_description.minor_version
-        interface_id = interface_description.interface_id
-        aggregation = interface_description.aggregation
-        storage = interface_description.storage
-        interface_type = interface_description.type
-        {:ok, mappings} = Queries.fetch_interface_mappings(conn, realm, interface_id, [])
-        mappings = Enum.sort_by(mappings, fn mapping -> mapping.endpoint end)
-
-        interface_attributes = [
-          interface_name: interface_name,
-          major_version: to_string(major_version),
-          minor_version: to_string(minor_version),
-          active: "true"
-        ]
-
-        type =
-          case interface_type do
-            :datastream ->
-              case aggregation do
-                :individual -> :individual
-                :object -> :object
-              end
-
-            :properties ->
-              :properties
-          end
-
-        [
-          %{
-            attributes: interface_attributes,
-            interface_id: interface_id,
-            device_id: device_id,
-            storage: storage,
-            type: type,
-            mappings: mappings
-          }
-          | acc
-        ]
-      end)
+      if introspection == nil do
+        []
+      else
+        Enum.reduce(introspection, [], fn {interface_name, major_version}, acc ->
```

Comment:
`Enum.reduce/3` on an empty list returns an empty list immediately. So you could write
```suggestion
      instrospection = if introspection == nil, do: [], else: introspection
      Enum.reduce(introspection, [], fn {interface_name, major_version}, acc ->
```
that allows to remove one indentation level